exonum/src/node/mod.rs

```diff
@@ -487,14 +487,25 @@ impl NodeHandler {
             info!("Trying to connect with peer {}", address);
         }
 
-        let round = Round::first();
+        let snapshot = self.blockchain.snapshot();
+        let schema = Schema::new(&snapshot);
+
+        // Recover previous saved round if any
+        let round = schema.consensus_round();
```

Comment:
Could there be any adverse effects, if say, the remembered round is 1000, and the other nodes are now at greater height with "normal" values of rounds?

Reply:
This isn't a problem: other nodes will ignore messages from the "previous" height. The restarted node firstly processes saved messages, which is not optimal, but makes code more maintainable because there is only one point for messages processing.